prow/tide/tide.go

```diff
@@ -346,8 +351,10 @@ func (c *Controller) Sync() error {
 	}
 	sortPools(pools)
 	c.m.Lock()
-	defer c.m.Unlock()
 	c.pools = pools
+	c.m.Unlock()
+
+	c.History.Flush()
```

Comment:
This shouldn't affect the queries at all. Queries are time bounded since the last time they started a successfully query (with some overlap), not since `now()-syncPeriod` so the sync loop taking a little extra time shouldn't have any impact on query correctness.